src/plots/gl3d/scene.js

```diff
@@ -456,10 +454,11 @@ proto.render = function() {
             gd.emit('plotly_hover', eventData);
         }
 
-        oldEventData = eventData;
+        this.oldEventData = eventData;
```

Comment:
What if we change this line to 
```js
if(eventData) this.oldEventData = eventData;
```
?

Reply:
I don't think this does the trick, unfortunately. Here's the logging output with this change.

```
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', Object{points: [Object{x: ..., y: ..., z: ..., data: ..., fullData: ..., curveNumber: ..., pointNumber: ..., marker.symbol: ..., marker.size: ..., marker.line.color: ..., marker.color: ..., bbox: ...}]}
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', undefined
```

`this.oldEventData` still gets set back to undefined after the `plotly_unhover` event has been emitted.

```javascript
gd.emit('plotly_unhover', this.oldEventData);    
this.oldEventData = undefined;
```

I could remove `this.oldEventData = undefined;` and then we could be pretty confident that the unhover event will have event data, as long as the user has hovered at least one point since graph creation (we'll still get undefined event data for renders before a point has been hovered). Here's the logging output.

```
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', undefined
LOG: 'UNHOVER', Object{points: [Object{x: ..., y: ..., z: ..., data: ..., fullData: ..., curveNumber: ..., pointNumber: ..., marker.symbol: ..., marker.size: ..., marker.line.color: ..., marker.color: ..., bbox: ...}]}
LOG: 'UNHOVER', Object{points: [Object{x: ..., y: ..., z: ..., data: ..., fullData: ..., curveNumber: ..., pointNumber: ..., marker.symbol: ..., marker.size: ..., marker.line.color: ..., marker.color: ..., bbox: ...}]}
LOG: 'UNHOVER', Object{points: [Object{x: ..., y: ..., z: ..., data: ..., fullData: ..., curveNumber: ..., pointNumber: ..., marker.symbol: ..., marker.size: ..., marker.line.color: ..., marker.color: ..., bbox: ...}]}
```

So this would solve the issue for the test case, but I think this behavior is misleading.